ssl-config/src/main/java/com/palantir/ssl/SslConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.ssl;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@JsonDeserialize(as = ImmutableSslConfiguration.class)
+@JsonSerialize(as = ImmutableSslConfiguration.class)
+@Value.Immutable
+@Value.Style(visibility = ImplementationVisibility.PACKAGE)
+public abstract class SslConfiguration {
```

Comment:
Yeah this is actually conceptually cleaner -- I actually had that implemented locally, but didn't push because the initial comments indicated a preference for a single factory :)

The single-instantiation case does get a bit more verbose (since, unless extra convenience constructors are added, even the minimal case will require 2 builders/objects -- one for the inner trust/identity and one for the outer config), but I do think it's cleaner.

I'll push up this implementation in another commit and people can provide feedback.